### PR TITLE
CircleCI v1 -> v2 

### PR DESCRIPTION
**Pull request by nathanleiby**
Issue: https://clever.atlassian.net/browse/INFRA-2404

Goal of this change is to start getting experience with CircleCI 2. 

Eventually, I'd like to:

- make our own base images which contain useful stuff (ci-scripts, aws cli, etc ... https://github.com/Clever/kinesis-to-firehose/blob/911cfaedd94d6691f009602e1c6810a664c2ad83/.circleci/config.yml#L10-L14)
- write a tool to automate moving from CCI 1 -> CCI 2

### CODE REVIEW

**Comment by nathanleiby** [comment]
(Ballpark) this seems to reduce build time from 5m -> 3m

**Comment by nathanleiby** [comment]
Build history: https://circleci.com/build-insights/gh/Clever/kinesis-to-firehose/master

vs this branch: https://circleci.com/build-insights/gh/Clever/kinesis-to-firehose/cci2

**Comment by nathanleiby** [comment]
(Note: this repo's Circle 2 config should be considered experimental, and we can try to work on best-practices/consistency among circle 2 config as we do a larger rollout.)

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+      - run: cd $HOME && git clone --depth 1 -v git@github.com:clever/ci-scripts.git && cd ci-scripts && git show --oneline -s
+      - run: make install_deps
+      - run: make build
+      - run: make test
+      - run: mkdir -p /tmp/test-results
+      - run: "make bench | tee /tmp/test-results/benchmarks.txt"
+      - store_artifacts:
+          path: /tmp/test-results
+          destination: benchmarks
+      - store_test_results:
```
Comment (mohit) [inline .circleci/config.yml]: I don't know what this means?

Comment (mohit) [reply .circleci/config.yml]: we should this remove this block. We don't have the right format, and also this will not work with CircleWorkflows.

**Review by mohit — APPROVED** [review]
one nit other than that 💯

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+      - checkout
+      - setup_remote_docker
+      - run: cd /tmp/ && wget https://bootstrap.pypa.io/get-pip.py && sudo python get-pip.py
+      - run: sudo pip install --upgrade awscli && aws --version
+      - run: pip install --upgrade --user awscli
+      - run: echo 'github.com ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEAq2A7hRGmdnm9tUDbO9IDSwBK6TbQa+PXYPCPy6rbTrTtw7PHkccKrpp0yVhp5HdEIcKr6pLlVDBfOLX9QUsyCOV0wzfjIJNlGEYsdlLJizHhbn2mUjvSAHQqZETYP81eFzLQNnPHt4EVVUh7VfDESU84KezmD5QlWpXLmvU31/yMf+Se8xhHTvKSCZIFImWwoG6mbUoWf9nzpIoaSjB+weqqUUmpaaasXVal72J+UX2B+2RPW3RcT0eOzQgqlJL3RKrTJvdsjE3JEAvGq3lGHSZXy28G3skua2SmVi/w4yCE6gbODqnTWlg7+wC604ydGXA8VJiS5ap43JXiUFFAaQ==' >> ~/.ssh/known_hosts
+      - run: cd $HOME && git clone --depth 1 -v git@github.com:clever/ci-scripts.git && cd ci-scripts && git show --oneline -s
+      - run: make install_deps
+      - run: make build
+      - run: make test
```
Comment (burnsed) [inline .circleci/config.yml:17]: wild that most sets are now just individual commands...

**Review by burnsed — APPROVED** [review]

**Pull request merged**